### ISSUE TITLE
Tailscale crashes on /e/OS 2.4.1 (Android 13)

### ISSUE THREAD

**Issue by Wojslav** [tailscale/tailscale#13923]
### What is the issue?

I have smartphone Samsung Galaxy S10e with /e/OS 2.4.1-t-20241009439852-community-beyound0lte with installed Tailscale 1.76.1-t24929f6b6-ga20b1114dd5. When I try to open Tailscale, it immediately closes. 

### Steps to reproduce

1. Turn on phone
2. Click on the Tailscale icon

### Are there any recent changes that introduced the issue?

I have recently updated my /e/OS, but I'm not sure if the app had started crashing only after update

### OS

Android

### OS version

/e/OS 2.4.1-t-20241009439852-community-beyound0lte (Android 13)

### Tailscale version

1.76.1-t24929f6b6-ga20b1114dd5

**Comment by bradfitz** [comment]
Where did you get your Android build? (Play Store, F-Droid, pkgs.tailcale.com download?)

Got any logs from `adb logcat` or anything while it crashes?

**Comment by thariq-shanavas** [comment]
Tailscale immediately crashes on open on my Pixel 9. App is on the latest Google Play version.
Here is the output of `adb logcat | grep tailscale`

```
10-27 21:23:25.265  1491  4447 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.tailscale.ipn/.MainActivity} with LAUNCH_SINGLE_TASK from uid 10353 (BAL_ALLOW_VISIBLE_WINDOW) result code=0
10-27 21:23:25.265  2087  2108 V WindowManagerShell: Transition requested (#4376): android.os.BinderProxy@105252a TransitionRequestInfo { type = OPEN, triggerTask = TaskInfo{userId=0 taskId=6047 displayId=0 isRunning=true baseIntent=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.tailscale.ipn/.MainActivity } baseActivity=ComponentInfo{com.tailscale.ipn/com.tailscale.ipn.MainActivity} topActivity=ComponentInfo{com.tailscale.ipn/com.tailscale.ipn.MainActivity} origActivity=null realActivity=ComponentInfo{com.tailscale.ipn/com.tailscale.ipn.MainActivity} numActivities=1 lastActiveTime=731471731 supportsMultiWindow=true resizeMode=1 isResizeable=true minWidth=-1 minHeight=-1 defaultMinSize=220 token=WCT{android.window.IWindowContainerToken$Stub$Proxy@8ffcf1b} topActivityType=1 pictureInPictureParams=null shouldDockBigOverlays=false launchIntoPipHostTaskId=-1 lastParentTaskIdBeforePip=-1 displayCutoutSafeInsets=Rect(0, 142 - 0, 0) topActivityInfo=ActivityInfo{fe798b8 com.tailscale.ipn.MainActivity} launchCookies=[] positionInParent=Point(0, 0) parentTaskId=-1 isFocused=false isVisible=false isVisibleRequested=false isSleeping=false locusId=null displayAreaFeatureId=1 isTopActivityTransparent=false appCompatTaskInfo=AppCompatTaskInfo { topActivityInSizeCompat=false topActivityEligibleForLetterboxEducation= falseisLetterboxEducationEnabled= false isLetterboxDoubleTapEnabled= false topActivityEligibleForUserAspectRatioButton= false topActivityBoundsLetterboxed= false isFromLetterboxDoubleTap= false topActivityLetterboxVerticalPosition= -1 topActivityLetterboxHorizontalPosition= -1 topActivityLetterboxWidth=1080 topActivityLetterboxHeight=2424 isUserFullscreenOverrideEnabled=false isSystemFullscreenOverrideEnabled=false cameraCompatTaskInfo=CameraCompatTaskInfo { cameraCompatControlState=hidden freeformCameraCompatMode=inactive}}}, pipTask = null, remoteTransition = null, displayChange = null, flags = 0, debugId = 4376 }
10-27 21:23:25.267 12225 12225 I TopTaskTracker: onTaskMovedToFront: (moved taskInfo to front) taskId=6047, baseIntent=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.tailscale.ipn/.MainActivity }
10-27 21:23:25.279  1491  4447 D CoreBackPreview: Window{42e5528 u0 Splash Screen com.tailscale.ipn}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@ad0217d, mPriority=0, mIsAnimationCallback=false}
10-27 21:23:25.280  1491  1738 I ActivityManager: Start proc 30057:com.tailscale.ipn/u0a372 for next-top-activity {com.tailscale.ipn/com.tailscale.ipn.MainActivity}
10-27 21:23:25.288 30057 30057 I m.tailscale.ipn: Using CollectorTypeCMC GC.
10-27 21:23:25.288  1491  1720 V WindowManager: Sent Transition (#4376) createdAt=10-27 21:23:25.258 via request=TransitionRequestInfo { type = OPEN, triggerTask = TaskInfo{userId=0 taskId=6047 displayId=0 isRunning=true baseIntent=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.tailscale.ipn/.MainActivity } baseActivity=ComponentInfo{com.tailscale.ipn/com.tailscale.ipn.MainActivity} topActivity=ComponentInfo{com.tailscale.ipn/com.tailscale.ipn.MainActivity} origActivity=null realActivity=ComponentInfo{com.tailscale.ipn/com.tailscale.ipn.MainActivity} numActivities=1 lastActiveTime=731471731 supportsMultiWindow=true resizeMode=1 isResizeable=true minWidth=-1 minHeight=-1 defaultMinSize=220 token=WCT{RemoteToken{d7a916c Task{cb2172f #6047 type=standard A=10372:com.tailscale.ipn}}} topActivityType=1 pictureInPictureParams=null shouldDockBigOverlays=false launchIntoPipHostTaskId=-1 lastParentTaskIdBeforePip=-1 displayCutoutSafeInsets=Rect(0, 142 - 0, 0) topActivityInfo=ActivityInfo{b070335 com.tailscale.ipn.MainActivity} launchCookies=[] positionInParent=Point(0, 0) parentTaskId=-1 isFocused=false isVisible=false isVisibleRequested=false isSleeping=false locusId=null displayAreaFeatureId=1 isTopActivityTransparent=false appCompatTaskInfo=AppCompatTaskInfo { topActivityInSizeCompat=false topActivityEligibleForLetterboxEducation= falseisLetterboxEducationEnabled= false isLetterboxDoubleTapEnabled= false topActivityEligibleForUserAspectRatioButton= false topActivityBoundsLetterboxed= false isFromLetterboxDoubleTap= false topActivityLetterboxVerticalPosition= -1 topActivityLetterboxHorizontalPosition= -1 topActivityLetterboxWidth=1080 topActivityLetterboxHeight=2424 isUserFullscreenOverrideEnabled=false isSystemFullscreenOverrideEnabled=false cameraCompatTaskInfo=CameraCompatTaskInfo { cameraCompatControlState=hidden freeformCameraCompatMode=inactive}}}, pipTask = null, remoteTransition = null, displayChange = null, flags = 0, debugId = 4376 }
10-27 21:23:25.288  1491  1720 V WindowManager:         {WCT{RemoteToken{d7a916c Task{cb2172f #6047 type=standard A=10372:com.tailscale.ipn}}} m=OPEN f=NONE leash=Surface(name=Task=6047)/@0xecca140 sb=Rect(0, 0 - 1080, 2424) eb=Rect(0, 0 - 1080, 2424) d=0 taskParent=-1},
10-27 21:23:25.290 30057 30057 E m.tailscale.ipn: Not starting debugger since process cannot load the jdwp agent.
10-27 21:23:25.307 30057 30057 D nativeloader: Configuring clns-7 for other apk /data/app/~~CIGxLcrZCfpklYe2OaoLQQ==/com.tailscale.ipn-rKqyEkyAeUdp9pl7vrWlgg==/base.apk. target_sdk_version=34, uses_libraries=, library_path=/data/app/~~CIGxLcrZCfpklYe2OaoLQQ==/com.tailscale.ipn-rKqyEkyAeUdp9pl7vrWlgg==/lib/arm64:/data/app/~~CIGxLcrZCfpklYe2OaoLQQ==/com.tailscale.ipn-rKqyEkyAeUdp9pl7vrWlgg==/base.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand:/data/user/0/com.tailscale.ipn
10-27 21:23:25.310 30057 30057 V GraphicsEnvironment: com.tailscale.ipn is not listed in per-application setting
10-27 21:23:25.331 30057 30072 D vulkan  : searching for layers in '/data/app/~~CIGxLcrZCfpklYe2OaoLQQ==/com.tailscale.ipn-rKqyEkyAeUdp9pl7vrWlgg==/lib/arm64'
10-27 21:23:25.331 30057 30072 D vulkan  : searching for layers in '/data/app/~~CIGxLcrZCfpklYe2OaoLQQ==/com.tailscale.ipn-rKqyEkyAeUdp9pl7vrWlgg==/base.apk!/lib/arm64-v8a'
10-27 21:23:25.343 30057 30057 D nativeloader: Load /data/app/~~CIGxLcrZCfpklYe2OaoLQQ==/com.tailscale.ipn-rKqyEkyAeUdp9pl7vrWlgg==/base.apk!/lib/arm64-v8a/libgojni.so using ns clns-7 from class loader (caller=/data/app/~~CIGxLcrZCfpklYe2OaoLQQ==/com.tailscale.ipn-rKqyEkyAeUdp9pl7vrWlgg==/base.apk): ok
10-27 21:23:25.348 30057     0 E Go      : panic: binary built with tailscale_go build tag but failed to read build info or find tailscale.toolchain.rev in build info
10-27 21:23:25.348 30057     0 E Go      : tailscale%2ecom.init.0()
10-27 21:23:25.348 30057     0 E Go      :      /home/vagrant/go/pkg/mod/tailscale.com@v1.76.0/assert_ts_toolchain_match.go:17 +0x190
10-27 21:23:25.348 30057 30091 E GoLog   : panic: binary built with tailscale_go build tag but failed to read build info or find tailscale.toolchain.rev in build info
10-27 21:23:25.348 30057 30088 F libc    : Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 30088 (m.tailscale.ipn), pid 30057 (m.tailscale.ipn)
10-27 21:23:25.478 30096 30096 F DEBUG   : Cmdline: com.tailscale.ipn
10-27 21:23:25.478 30096 30096 F DEBUG   : pid: 30057, tid: 30088, name: m.tailscale.ipn  >>> com.tailscale.ipn <<<
10-27 21:23:25.478 30096 30096 F DEBUG   :       #00 pc 00000000004b1bf8  /data/app/~~CIGxLcrZCfpklYe2OaoLQQ==/com.tailscale.ipn-rKqyEkyAeUdp9pl7vrWlgg==/base.apk (offset 0x2ac000)
10-27 21:23:25.487  1491 30099 W ActivityTaskManager:   Force finishing activity com.tailscale.ipn/.MainActivity
10-27 21:23:25.507  1491  4447 I ActivityManager: Process com.tailscale.ipn (pid 30057) has died: fg  TOP 
10-27 21:23:25.521  1491  1720 V WindowManager:         {WCT{RemoteToken{d7a916c Task{cb2172f #6047 type=standard A=10372:com.tailscale.ipn}}} m=CLOSE f=NONE leash=Surface(name=Task=6047)/@0xecca140 sb=Rect(0, 0 - 1080, 2424) eb=Rect(0, 0 - 1080, 2424) d=0 taskParent=-1}
10-27 21:23:25.529  1491  2260 D CoreBackPreview: Window{42e5528 u0 Splash Screen com.tailscale.ipn EXITING}: Setting back callback null
10-27 21:23:25.530  1491  2260 W InputManager-JNI: Input channel object '42e5528 Splash Screen com.tailscale.ipn (client)' was disposed without first being removed with the input manager!
10-27 21:23:25.988  1491  1723 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{cf9f10e u0 com.tailscale.ipn/.MainActivity t-1 f}}

```

**Comment by bradfitz** [comment]
> App is on the latest Google Play version

That's not true; you're using the F-Droid version:

```
10-27 21:23:25.348 30057     0 E Go      : panic: binary built with tailscale_go build tag but failed to read build info or find tailscale.toolchain.rev in build info
10-27 21:23:25.348 30057     0 E Go      : tailscale%2ecom.init.0()
10-27 21:23:25.348 30057     0 E Go      :      /home/vagrant/go/pkg/mod/tailscale.com@v1.76.0/assert_ts_toolchain_match.go:17 +0x190
10-27 21:23:25.348 30057 30091 E GoLog   : panic: binary built with tailscale_go build tag but failed to read build info or find tailscale.toolchain.rev in build info
10-27 21:23:25.348 30057 30088 F libc    : Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 30088 (m.tailscale.ipn), pid 30057 (m.tailscale.ipn)
```

We don't build the official binaries with Vagrant. F-Droid does.

This is a duplicate of #13850 

I recommend you uninstall the F-Droid version and actually use the Google Play version, or the official APK from https://pkgs.tailscale.com/stable/#android

**Comment by thariq-shanavas** [comment]
My apologies. It was indeed installed from F-droid on my phone - it was a mix-up on my part.
The Google play version seems to be working well. Thank you so much!

**Comment by Wojslav** [comment]
I've got it from F-Droid (not directly - it's available through App Lounge, a default app store in /e/ OS).

From `adb logcat`, I've got such logs:

`10-28 16:14:42.192 16573 16573 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
10-28 16:14:42.192 16573 16573 F DEBUG   : LineageOS Version: '2.4.1-t-20241009439852-community-beyond0lte'
10-28 16:14:42.192 16573 16573 F DEBUG   : Build fingerprint: 'samsung/lineage_beyond0lte/beyond0lte:13/TQ3A.230901.001/eng.root.20241009.020129:userdebug/release-keys'
10-28 16:14:42.192 16573 16573 F DEBUG   : Revision: '26'
10-28 16:14:42.192 16573 16573 F DEBUG   : ABI: 'arm64'
10-28 16:14:42.192 16573 16573 F DEBUG   : Timestamp: 2024-10-28 16:14:41.757719231+0100
10-28 16:14:42.192 16573 16573 F DEBUG   : Process uptime: 1s
10-28 16:14:42.192 16573 16573 F DEBUG   : Cmdline: com.tailscale.ipn
10-28 16:14:42.192 16573 16573 F DEBUG   : pid: 16542, tid: 16563, name: m.tailscale.ipn  >>> com.tailscale.ipn <<<
10-28 16:14:42.192 16573 16573 F DEBUG   : uid: 10204
10-28 16:14:42.192 16573 16573 F DEBUG   : signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
10-28 16:14:42.192 16573 16573 F DEBUG   :     x0  0000000000000000  x1  00000000000040b3  x2  0000000000000006  x3  0000000000000008
10-28 16:14:42.192 16573 16573 F DEBUG   :     x4  0000000000000001  x5  0000000000000000  x6  0000000000000000  x7  0000000000000001
10-28 16:14:42.192 16573 16573 F DEBUG   :     x8  0000000000000083  x9  000000000000db7c  x10 000000000000000e  x11 00000077e2912edc
10-28 16:14:42.192 16573 16573 F DEBUG   :     x12 0000000000000001  x13 00000077e1cfb2c4  x14 00000077e1d63cb1  x15 000000000000000c
10-28 16:14:42.192 16573 16573 F DEBUG   :     x16 00000077e1840f30  x17 00000077e19375d0  x18 00000077e157a000  x19 000000000000409e
10-28 16:14:42.192 16573 16573 F DEBUG   :     x20 00000077e1937470  x21 00000077e2f7ee40  x22 0000000000000013  x23 0000003cb9d69707
10-28 16:14:42.192 16573 16573 F DEBUG   :     x24 8e41dc0be9fcd734  x25 000000be71521dac  x26 0000000000000000  x27 0000000000000010
10-28 16:14:42.192 16573 16573 F DEBUG   :     x28 00000040000021c0  x29 0000004000080c58
10-28 16:14:42.192 16573 16573 F DEBUG   :     lr  00000077e1dc5584  sp  0000004000080c60  pc  00000077e1dedc28  pst 0000000080000000
10-28 16:14:42.192 16573 16573 F DEBUG   : backtrace:
10-28 16:14:42.192 16573 16573 F DEBUG   :       #00 pc 00000000004b1c28  /data/app/~~3syPJW5ew0rhKkDYi7bZpw==/com.tailscale.ipn-nM8rw21ixIhUnbIH9cb1Eg==/base.apk (offset 0x2ac000)
`

**Comment by bradfitz** [comment]
@Wojslav, that's the same thing. The F-Droid build was buggy and untested. https://github.com/tailscale/tailscale/issues/13850 tracks the fix efforts.
